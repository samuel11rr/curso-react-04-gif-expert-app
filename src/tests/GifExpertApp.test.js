import React from 'react'

const { shallow } = require("enzyme");
const { GifExpertApp } = require("../GifExpertApp");

describe('Pruenas en <GifExpertApp />', () => {
  test('Debe mostrarse correctamente', () => {
    const wrapper = shallow( <GifExpertApp /> );
    expect( wrapper ).toMatchSnapshot();
  });
  

  test('Debe mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Dragon Ball'];

    const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
  });
  
});
