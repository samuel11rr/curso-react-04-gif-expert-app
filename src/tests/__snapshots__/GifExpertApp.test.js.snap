// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruenas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    newCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruenas en <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    newCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
